rlog_test.go

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package rlog
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+)
+
+var logfile string
+var removeLogfile bool = false
+var fixedLogfileName bool = true
+
+// setup is called at the start of each test and prepares a new log file. It
+// also returns a new configuration, which can be used by this test.
+func setup() rlogConfig {
+	if fixedLogfileName {
+		logfile = "/tmp/rlog-test.log"
+	} else {
+		logfile = fmt.Sprintf("/tmp/rlog-test-%d.log", time.Now().UnixNano())
+	}
+	os.Remove(logfile)
+
+	return rlogConfig{
+		logLevel:       "",
+		traceLevel:     "",
+		logTimeFormat:  "",
+		logFile:        logfile,
+		logStream:      "NONE",
+		logTimeDate:    false,
+		showCallerInfo: false,
+	}
+}
+
+// cleanup is called at the end of each test.
+func cleanup() {
+	if removeLogfile {
```

Comment:
I think the idea is that the test can set it to true (and also change fixedLogfileName as above)